test/socket/main.cpp

```diff
@@ -36,7 +36,7 @@ std::string subj(const char* name) {
 }
 
 TEST_CASE( "Set/Get options", "[socket]" ) {
-    boost::asio::io_service ios;
+    boost::asio::io_context ios;
```

Comment:
Would name this ioc instead of ios to be fully up to date.

Reply:
newer boost version provides an alias for io_service -> io_context so this shouldn't be an issues

Reply:
Yeah they have. Anyway it does not make the code more readable when everything gets mixed up. Stick to a clean line and follow that one should not hurt that much and makes it more easy to get beginners into this library.

Reply:
I'll need to review when io-context came in, accepting this change would lift the minimum version of boost azmq requires.

Reply:
io_context was introduced in boost 1.66.0, and at the moment we claim support back to 1.48.0